docs/delivery/README.md

```diff
@@ -0,0 +1,121 @@
+# Delivery Design
+
+This document synthetizes the [error handling design document](https://docs.google.com/document/d/1qRrzGoHJQO-oc5p-yRK8IRfugd-FM_PXyM7lN5kcqks).
+
+## Problem
+
+Sending events can fail for a variety of reasons (downstream system is down, application logic rejects the invalid message, a runtime exception occurs, etc...) but right now there is no way to control or define the expected behavior in these situations.
+
+## Requirements
+
+* Be able to handle events that failed to be delivered
+    * to channel subscribers.
+    * to source sink.
+    * to broker/triggers.
+* Be able to identify a message couldn’t be delivered (Observability)
+* Be able to leverage existing native error handling mechanisms (eg. dead letter queues).
+* Be able to redirect of error'ed events from a channel.
```

Comment:
events? or messages?  let's be consistent, and I think `event` is the right term, over `message(s)` /cc @nachocano

Reply:
I'm fine with using event everywhere.